adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java

```diff
@@ -482,12 +487,13 @@ public TerminalAPIRequest createTerminalAPIRequest(final CartData cartData, Cust
 
         paymentRequest.setPaymentTransaction(paymentTransaction);
 
-        saleToPOIRequest.setPaymentRequest(paymentRequest);
-
-        TerminalAPIRequest terminalApiRequest = new TerminalAPIRequest();
-        terminalApiRequest.setSaleToPOIRequest(saleToPOIRequest);
+        TerminalAPIRequestBuilder builder = new TerminalAPIRequestBuilder()
+                .withSaleId(cartData.getStore())
+                .withServiceId(serviceId)
+                .withPoiId(cartData.getAdyenTerminalId())
```

Comment:
those are removed from the library and need to be passed into the constructor

Reply:
fixed